### PR TITLE
Moved pingers to seperate file

### PR DESCRIPTION
**Pull request by HermanPlay**
- Implemented generic pinger, which takes callback function to ping and check if ping is successful, third function is called when error happens during ping. 

- Fixed issues with previous commits, where tried to `raise JSONResponse`. 

- Changed deploy, deactivate, and build to run asynchronously

### CODE REVIEW

**Review by matined — CHANGES_REQUESTED** [review]
The code from DeploymentPinger is still duplicated in ServicePinger ;) What about a generic function that takes an object and a method like `read_namespaced_deployment_status` as a argument?

File: app/api/utils/cluster_pingers.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import time
+
+from utils.exception import PingLimitReached, ErrImagePull, EmptyList
+
+_logger = logging.getLogger(__name__)
+
+
+class Pinger:
+    """
+    Pings services in a given namespace using kubernetes api.
```
Comment (matined) [inline app/api/utils/cluster_pingers.py:13]: There is 'service' in all the docstrings, it it correct?

File: app/api/utils/cluster_pingers.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    """
+
+    def __init__(
```
Comment (matined) [inline app/api/utils/cluster_pingers.py:16]: This set of parameters makes the method not really generic, I'd remove name and namespace and put:

ping_callback: Callable
ping_callback_args: list | dict
ping_predicate: Callable
error_callback: Callable
error_callback_args: list | dict
ping_limit: int

File: app/api/utils/cluster_pingers.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    @classmethod
+    def deployment_ping(cls, name: str, namespace: str) -> V1PodList:
+        """
+        Returns a list of pods for a given deployment
+
+        :param name: Name of the deployment
+        :param namespace: Namespace of the deployment
+        :return: List of pods
+        """
+        config.load_incluster_config()
```
Comment (matined) [inline app/api/utils/cluster_pingers.py:90]: Already in constructor

**Review by matined — CHANGES_REQUESTED** [review]

File: app/api/utils/cluster_pingers.py (hunk 1 above)
Comment (HermanPlay) [reply app/api/utils/cluster_pingers.py:13]: Yes, but i don't mean kubernetes [service](https://kubernetes.io/docs/concepts/services-networking/service/), rather i mean anything which is in the cluster, deployment, pod, role, etc.

File: app/api/utils/cluster_pingers.py (hunk 3 above)
Comment (HermanPlay) [reply app/api/utils/cluster_pingers.py:90]: Done

File: app/api/utils/cluster_pingers.py (hunk 1 above)
Comment (matined) [reply app/api/utils/cluster_pingers.py:13]: can we use 'resource' not to mislead?

**Review by matined — APPROVED** [review]

**Comment by matined** [comment]
Good job, let's merge it, we'll fix bugs if there are ones later ;)

**Pull request merged**